src/dex/fluid-dex/fluid-dex.ts

```diff
@@ -187,42 +187,53 @@ export class FluidDex extends SimpleExchange implements IDex<FluidDexData> {
     try {
       if (srcToken.address.toLowerCase() === destToken.address.toLowerCase())
         return null;
-
-      if (side === SwapSide.BUY) return null;
       // Get the pool to use.
       const pool = this.getPoolByTokenPair(srcToken.address, destToken.address);
       if (!pool) return null;
-
       // Make sure the pool meets the optional limitPools filter.
       if (limitPools && !limitPools.includes(pool.id)) return null;
 
       const liquidityProxyState = await this.liquidityProxy.getStateOrGenerate(
         blockNumber,
       );
-
       const currentPoolReserves = liquidityProxyState.poolsReserves.find(
         poolReserve =>
           poolReserve.pool.toLowerCase() === pool.address.toLowerCase(),
       );
-      if (!currentPoolReserves) return null;
-
+      if (!currentPoolReserves) {
+        return null;
+      }
       const prices = amounts.map(amount => {
-        return this.swapIn(
-          srcToken.address.toLowerCase() === pool.token0.toLowerCase(),
-          amount,
-          currentPoolReserves.collateralReserves,
-          currentPoolReserves.debtReserves,
-          srcToken.decimals,
-          destToken.decimals,
-          BigInt(currentPoolReserves.fee),
-        );
+        if (side === SwapSide.SELL) {
+          return this.swapIn(
+            srcToken.address.toLowerCase() === pool.token0.toLowerCase(),
+            amount,
+            currentPoolReserves.collateralReserves,
+            currentPoolReserves.debtReserves,
+            srcToken.decimals,
+            destToken.decimals,
+            BigInt(currentPoolReserves.fee),
+            currentPoolReserves.dexLimits,
+            Math.floor(Date.now() / 1000),
+          );
+        } else {
+          return this.swapOut(
+            srcToken.address.toLowerCase() === pool.token0.toLowerCase(),
+            amount,
+            currentPoolReserves.collateralReserves,
+            currentPoolReserves.debtReserves,
+            srcToken.decimals,
+            destToken.decimals,
+            BigInt(currentPoolReserves.fee),
+            currentPoolReserves.dexLimits,
+            Math.floor(Date.now() / 1000),
+          );
+        }
       });
       return [
         {
           prices: prices,
-          unit: getBigIntPow(
-            (side === SwapSide.SELL ? destToken : srcToken).decimals,
-          ),
+          unit: getBigIntPow(destToken.decimals),
```

Comment:
is this correct?

Reply:
@JackieJoo unit is not used anymore. I think anything could be passed here.